File: ScaleformUI_Csharp/Items/UIMenuProgressItem.cs
using CitizenFX.Core;
using CitizenFX.Core.Native;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScaleformUI
{
	public class UIMenuProgressItem : UIMenuItem
	{
		protected internal bool Pressed;
		protected internal UIMenuGridAudio Audio;

		protected internal int _value = 0;
		protected internal int _max = 100;
		protected internal int _multiplier = 5;
        private HudColor sliderColor;
		protected internal bool Divider;
		protected internal HudColor SliderColor
		{
			get => sliderColor;
			set
			{
				sliderColor = value;
				if (Parent is not null && Parent.Visible)
				{
					ScaleformUI._ui.CallFunction("UPDATE_COLORS", Parent.MenuItems.IndexOf(this), (int)MainColor, (int)HighlightColor, (int)TextColor, (int)HighlightedTextColor, (int)value);
				}
			}
		}

		public UIMenuProgressItem(string text, int maxCount, int startIndex) : this(text, maxCount, startIndex, "")
		{
			_max = maxCount;
			_value = startIndex;
		}

		public UIMenuProgressItem(string text, int maxCount, int startIndex, string description, bool heritage = false) : this(text, maxCount, startIndex, description, HudColor.HUD_COLOUR_FREEMODE)
		{
		}

		public UIMenuProgressItem(string text, int maxCount, int startIndex, string description, HudColor sliderColor) : base(text, description)
		{
			_max = maxCount;
			_value = startIndex;
			SliderColor = sliderColor;
			Audio = new UIMenuGridAudio("CONTINUOUS_SLIDER", "HUD_FRONTEND_DEFAULT_SOUNDSET", 0);
			_itemId = 4;
		}

		public int Value
		{
			get
			{
				return _value;
			}
			set
			{
				if (value > _max)
					_value = _max;
				else if (value < 0)
					_value = 0;
				else
					_value = value;
				ProgressChanged(Value);
			}
		}

		public int Multiplier
		{
			get => _multiplier;
			set =>_multiplier = value;
		}

		/// <summary>
		/// Triggered when the slider is changed.
		/// </summary>
		public event ItemSliderProgressEvent OnSliderChanged;

		internal virtual void ProgressChanged(int value)
		{
			OnSliderChanged?.Invoke(this, value);
		}

		public override void SetRightBadge(BadgeIcon badge)
		{
			throw new Exception("UIMenuProgressItem cannot have a right badge.");
		}

		public override void SetRightLabel(string text)
		{
			throw new Exception("UIMenuProgressItem cannot have a right label.");
		}
	}
}

File: ScaleformUI_V8/src/menus/UIMenu/items/uimenuprogressitem.ts
import { SColor } from "elements/scolor";
import { UIMenuItem } from "./uimenuitem";
import { UIMenuGridAudio } from "elements/uimenugridaudio";
import { ItemSliderProgressCallback, ProgressItemCallbackBuilder } from "../emitters/emitters";
import { BadgeStyle } from "elements/badge";

export class UIMenuProgressItem extends UIMenuItem {
    Pressed: boolean;
    Audio: UIMenuGridAudio;
    _value: number = 0;
    _max: number = 100;
    _multiplier: number = 5;
    sliderColor: SColor;
    Divider: boolean = false;
    _itemSliderProgressCallback = new ProgressItemCallbackBuilder()


    constructor(text: string, maxCount: number, startIndex: number, description: string, sliderColor: SColor) {
        super(text, description);
        this._max = maxCount;
        this._value = startIndex;
        this.SliderColor = sliderColor;
        this.Audio = new UIMenuGridAudio("CONTINUOUS_SLIDER", "HUD_FRONTEND_DEFAULT_SOUNDSET", 0);
        this._itemId = 4;
        this.Pressed = false;
    }

    public set SliderColor(value: SColor) {
        this.sliderColor = value;
        if (this.Parent !== null && this.Parent.Visible && this.Parent.Pagination.IsItemVisible(this.Parent.Items.indexOf(this))) {
            ScaleformUI.Scaleforms._ui.callFunction("UPDATE_COLORS", this.Parent.Pagination.GetScaleformIndex(this.Parent.Items.indexOf(this)), this.MainColor.toArgb(), this.HighlightColor.toArgb(), this.TextColor, this.HighlightedTextColor, value);
        }
    }
    public get SliderColor(): SColor {
        return this.sliderColor;
    }

    public set Value(value: number) {
        if (value > this._max)
            this._value = this._max;
        else if (value < 0)
            this._value = 0;
        else
            this._value = value;
        this.progressChanged(this._value)
        if (this.Parent !== null && this.Parent.Visible && this.Parent.Pagination.IsItemVisible(this.Parent.Items.indexOf(this))) {
            ScaleformUI.Scaleforms._ui.callFunction("SET_ITEM_VALUE", this.Parent.Pagination.GetScaleformIndex(this.Parent.Items.indexOf(this)), this._value);
        }
    }
    public get Value(): number {
        return this._value;
    }

    public set Multiplier(value: number) {
        this._multiplier = value;
    }
    public get Multiplier(): number {
        return this._multiplier;
    }

    public onProgressChanged(delegate: ItemSliderProgressCallback) {
        this._itemSliderProgressCallback.add(delegate)
    }

    progressChanged(value: number) {
        this._itemSliderProgressCallback.toDelegate()(this, value);
    }

    public SetRightBadge(badge: BadgeStyle) {
        throw new Error("UIMenuProgressItem cannot have a right badge.");
    }

    public SetRightLabel(text: string) {
        throw new Error("UIMenuProgressItem cannot have a right label.");
    }
}